dotnet/samples/Concepts/Memory/VectorStoreEmbeddingGeneration/TextEmbeddingVectorStoreExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.VectorData;
+using Microsoft.SemanticKernel.Embeddings;
+
+namespace Memory.VectorStoreEmbeddingGeneration;
+
+/// <summary>
+/// Contains extension methods to help add text embedding generation to a <see cref="IVectorStore"/> or <see cref="IVectorStoreRecordCollection{TKey, TRecord}"/>
+/// </summary>
+/// <remarks>
+/// This class is part of the <see cref="VectorStore_EmbeddingGeneration"/> sample.
+/// </remarks>
+public static class TextEmbeddingVectorStoreExtensions
+{
+    /// <summary>
+    /// Add text embedding generation to a <see cref="IVectorStore"/>.
+    /// </summary>
+    /// <param name="vectorStore">The <see cref="IVectorStore"/> to add text embedding generation to.</param>
+    /// <param name="textEmbeddingGenerationService">The service to use for generating text embeddings.</param>
+    /// <returns>The <see cref="IVectorStore"/> with text embedding added.</returns>
+    public static IVectorStore WithTextEmbeddingGeneration(this IVectorStore vectorStore, ITextEmbeddingGenerationService textEmbeddingGenerationService)
```

Comment:
In Microsoft.Extensions.AI their naming has `UseXXX` for example [OpenTelemetryChatClientBuilderExtensions.cs](https://github.com/dotnet/extensions/blob/main/src/Libraries/Microsoft.Extensions.AI/ChatCompletion/OpenTelemetryChatClientBuilderExtensions.cs). Should we follow the same pattern?

Reply:
Makes sense. Updated. Thanks.